### ISSUE TITLE
Problem prediction hurdle model

### ISSUE THREAD

**Issue by naveranoc** [glmmTMB/glmmTMB#860]
Hi, 

Looking at the package and with the help of this page https://data.princeton.edu/wws509/notes/countmoments, there seems to be a bug with hurdle model prediction using type = "response".

Here is an example showing the error:

```r
library(glmmTMB)
data(Salamanders)
Salamanders = transform(Salamanders, present = as.numeric(count>0))
hpm1 = glmmTMB(formula = count~mined,
               ziformula = ~mined,
               data =  Salamanders, 
               family=truncated_poisson)
predict(hpm1,type = "response")[1:5]
pred_cond = predict(hpm1,type = "cond")
pred_zero = predict(hpm1,type = "zprob")
mu_final = (1-pred_zero)*pred_cond/(1-exp(-pred_cond))
mu_final[1:5]
predict(hpm1,type = "response")[1:5]
```

Here I leave the comparison that I get from the values

```r
> mu_final[1:5] # manual calculated value
[1] 0.2954545 0.2954545 0.2954545 2.2648811 2.2648811
```

```r
> predict(hpm1,type = "response")[1:5] # package calculated value
[1] 0.2224856 0.2224856 0.2224856 2.2000815 2.2000815`
```

Thank you very much for your help

**Comment by mebrooks** [comment]
I think this is the same as issue #634

**Comment by bbolker** [comment]
On the `truncated_predict_fix` branch, `type="conditional"` now returns the mean of the *truncated* Poisson (as I think it should)  and

```r
all.equal((1-pred_zero)*pred_cond, predict(hpm1, type = "response"))
```
is TRUE. Can you please confirm that makes sense? I think

```r
remotes::install_github("glmmTMB/glmmTMB/glmmTMB@truncated_predict_fix")
```

should install the fixed branch (but `remotes::install_github()` isn't working for me at all at the moment ...

**Comment by bbolker** [comment]
should be fixed by 1e37cbecdbe607 (feel free to reopen if there are further issues)
